servicetalk-opentracing-zipkin-publisher/src/main/java/io/servicetalk/opentracing/zipkin/publisher/reporter/HttpReporter.java

```diff
@@ -0,0 +1,287 @@
+        final PublisherSource.Processor<Span, Span> buffer;
+        SpanBytesEncoder spanEncoder = builder.codec.spanBytesEncoder();
+        final BufferAllocator allocator = client.executionContext().bufferAllocator();
+        final Publisher<Buffer> spans;
+        if (builder.disableBatching) {
+            buffer = newPublisherProcessorDropHeadOnOverflow(builder.maxConcurrentReports);
+            spans = fromSource(buffer).map(span -> allocator.wrap(spanEncoder.encode(span)));
+        } else {
+            // As we send maxConcurrentReports number of parallel requests, each with roughly batchSizeHint number of
+            // spans, we hold a maximum of that many Spans in-memory that we can send in parallel to the collector.
+            buffer = newPublisherProcessorDropHeadOnOverflow(builder.batchSizeHint * builder.maxConcurrentReports);
+            spans = fromSource(buffer)
+                    .buffer(forCountOrTime(builder.batchSizeHint, builder.maxBatchDuration,
+                            () -> new ListAccumulator(builder.batchSizeHint), client.executionContext().executor()))
+                    .filter(accumulate -> !accumulate.isEmpty())
+                    .map(bufferedSpans -> allocator.wrap(spanEncoder.encodeList(bufferedSpans)));
+        }
+
+        final CompletableSource.Processor spansTerminated = newCompletableProcessor();
+        toSource(spans.flatMapCompletable(encodedSpansReporter(client, builder.codec), builder.maxConcurrentReports))
+                .subscribe(spansTerminated);
+
+        closeable.prepend(toAsyncCloseable(graceful -> {
+            closeInitiated = true;
+            try {
+                buffer.onComplete();
+            } catch (Throwable t) {
+                LOGGER.error("Failed to dispose request buffer. Ignoring.", t);
+            }
+            return graceful ? fromSource(spansTerminated) : completed();
+        }));
+
+        return buffer;
+    }
+
+    private static Function<Buffer, Completable> encodedSpansReporter(final HttpClient client, final Codec codec) {
+        final String path;
+        final CharSequence contentType;
+        switch (codec) {
+            case JSON_V1:
+                path = V1_PATH;
+                contentType = APPLICATION_JSON;
+                break;
+            case JSON_V2:
+                path = V2_PATH;
+                contentType = APPLICATION_JSON;
+                break;
+            case THRIFT:
+                path = V2_PATH;
+                contentType = THRIFT_CONTENT_TYPE;
+                break;
+            case PROTO3:
+                path = V2_PATH;
+                contentType = PROTO_CONTENT_TYPE;
+                break;
+            default:
+                throw new IllegalArgumentException("Unknown codec: " + codec);
+        }
+        return encodedSpans -> client.request(
+                client.post(path).addHeader(CONTENT_TYPE, contentType).payloadBody(encodedSpans))
+                .ignoreElement()
```

Comment:
Before we `ignoreElement`, should we log (at warn or error level) any response statuses that are not `SUCCESSFUL_2XX`?

Reply:
Good point, lemme add a  log.